pkg/odo/cli/service/create.go

```diff
@@ -45,46 +49,96 @@ func NewServiceCreateOptions() *ServiceCreateOptions {
 
 // Complete completes ServiceCreateOptions after they've been created
 func (o *ServiceCreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	o.serviceType = args[0]
-	// if only one arg is given, then it is considered as service name and service type both
-	o.serviceName = o.serviceType
-	// if two args are given, first is service type and second one is service name
-	if len(args) == 2 {
-		o.serviceName = args[1]
+	if len(args) == 0 || !cmd.HasFlags() {
+		o.interactive = true
 	}
+
 	o.Context = genericclioptions.NewContextCreatingAppIfNeeded(cmd)
-	return err
+
+	client := o.Client
+
+	var class scv1beta1.ClusterServiceClass
```

Comment:
With regards to this code section below, is there any way to refactor this into a separate file or function? The only reason being that `create.go` is becoming extremely large now..

Reply:
I could indeed split this into several functions but I think the code should be kept in this file and this would not really improve the size of the file.